aixtend/utils/file_handler.py

```diff
@@ -0,0 +1,43 @@
+"""
+Copyright 2022 The aiXplain SDK authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+
+import tempfile
+from pathlib import Path
+from uuid import uuid4
+import requests
+
+def download_file(download_url: str, download_file_path=None) -> str:
+    """Download and save file from given URL
+
+    Args:
+        download_url (str): URL of file to download
+        download_file_path (str, optional): File path to save downloaded file. If None then generates random path is tempdirectory. Defaults to None.
+
+    Returns:
+        str: Path where file was downloaded
+    """
+    if download_file_path is None:
+        sys_temp_dir = tempfile.gettempdir()
```

Comment:
Is there a way to have the temp file persist even after code execution finishes? A user will probably want to access the file later as well. The download path will now be something like '/tmp/aiXtend/filename.ext'. Do you have suggestions on if we should save it someplace else

Reply:
Tempdir and tempfile are both the same in terms of retaining.
Why do you see that a download utility function needs the file to be retained after it is processed? A user can download it again if they want to check the contents. Whereas a library has to be cautious about disk usage and cache items.

Reply:
tempfile.gettempdir() just gives the default temp folder of the machine and doesn't actually create a temporary directory or file.
 This is for the functionality of downloading the results of the benchmarks as a CSV so it needs to be persistent. In case a save path isn't provided we use this path.